orangecontrib/text/twitter.py

```diff
@@ -0,0 +1,257 @@
+
+    def join(self, *args):
+        if self.task:
+            self.task.join(*args)
+
+    def add_status(self, status):
+        status_record = {attr.name: getter(status)
+                         for attr, getter in self.supported_fields}
+        self.statuses_lock.acquire()
+        self.container[status.id] = status_record
+        self.statuses_lock.release()
+
+    def create_corpus(self, included_attributes=None):
+        """ Creates a corpus with collected tweets. """
+        if included_attributes:
+            attributes = [(attr, _) for attr, _ in self.attributes
+                          if attr.name in included_attributes]
+
+            metas = [(attr, _) for attr, _ in self.metas
+                     if attr.name in included_attributes]
+
+            text_features = [attr for attr in self.text_features
+                             if attr.name in included_attributes]
+        else:
+            attributes = self.attributes
+            metas = self.metas
+            text_features = self.text_features
+
+        domain = data.Domain(attributes=[attr for attr, _ in attributes],
+                             metas=[attr for attr, _ in metas])
+
+        self.statuses_lock.acquire()
+
+        for attr in domain.attributes:
+            if isinstance(attr, data.DiscreteVariable):
+                attr.values = []
+
+        def to_val(attr, val):
+            if isinstance(attr, data.DiscreteVariable) and val not in attr.values:
+                attr.add_value(val)
+            return attr.to_val(val)
+
+        X = np.array([
+            [to_val(attr, record[attr.name]) for attr, _ in attributes]
+            for record in self.tweets
+        ])
+
+        metas = np.array([
+            [record[attr.name] for attr, _ in metas]
+            for record in self.tweets
+        ], dtype=object)
+        self.statuses_lock.release()
+
+        return Corpus(X=X, metas=metas, domain=domain, text_features=text_features)
+
+    def reset(self):
+        """ Removes all downloaded tweets. """
+        if self.task:
+            self.task.disconnect()
+            self.task.join()
+        self.history = []
+        self.container = OrderedDict()
+
+
+class SearchTask(threading.Thread):
+    def __init__(self, master, q, lang=None, max_tweets=None, **kwargs):
+        super().__init__()
+        self.master = master
+        self.q = q
+        self.lang = lang
+        self.running = False
+        self.max_tweets = max_tweets
+        self.kwargs = kwargs
+
+    def disconnect(self):
+        self.running = False
+
+    def start(self):
+        self.running = True
+        self.progress = 0
+        if self.master.on_start:
+            self.master.on_start()
+        super().start()
+
+    def run(self):
+        try:
+            for status in tweepy.Cursor(self.master.api.search, q=self.q,
+                                        lang=self.lang, **self.kwargs).items(self.max_tweets):
```

Comment:
This is causing the problems when `self.max_tweets = None`. We should probably pass `0` when self.max_tweets is None.